R/gs_update_ahr.R

```diff
@@ -153,83 +83,48 @@
+#'   ustime = c(190 / planned_event_fa, 1),
+#'   lstime = c(190 / planned_event_fa, 1),
+#'   event_tbl = data.frame(analysis = c(1, 1),
+#'                          event = c(50, 140)))
 #'
-#' # Example B4 ----
-#' # IA spending = min(observed events, planned events) / final planned events
-#' ustime <- c(min(observed_event_ia, planned_event_ia),
-#'             min(observed_event_fa, planned_event_fa)) / planned_event_fa
+#' # Updated design with 190 events observed at IA, and 300 events observed at FA,
+#' # where 50 events observed during the delayed effect.
+#' # IA spending = observed events / final planned events, the remaining alpha will be allocated to FA.
 #' gs_update_ahr(
 #'   x = x,
-#'   ustime = ustime,
-#'   lstime = ustime,
-#'   observed_data = list(observed_data_ia, observed_data_fa))
+#'   ustime = c(190 / planned_event_fa, 1),
+#'   lstime = c(190 / planned_event_fa, 1),
+#'   event_tbl = data.frame(analysis = c(1, 1, 2, 2),
+#'                          event = c(50, 140, 50, 250)))
 #'
-#' # Example B5 ----
-#' # alpha is updated to 0.05 ----
-#' gs_update_ahr(x = x, alpha = 0.05)
-#'
-#' # Example B6 ----
-#' # updated boundaries only when IA data is observed
-#' ustime <- c(observed_event_ia / planned_event_fa, 1)
+#' # Updated design with 190 events observed at IA, and 300 events observed at FA,
+#' # where 50 events observed during the delayed effect.
+#' # IA spending = minimal of planned and actual information fraction spending
 #' gs_update_ahr(
 #'   x = x,
-#'   ustime = ustime,
-#'   lstime = ustime,
-#'   observed_data = list(observed_data_ia, NULL))
-#'
-#' # ------------------------------------------------- #
-#' # Example C: Two-sided asymmetric design,
-#' # with calendar spending for efficacy and futility bounds
-#' # beta-spending with non-binding lower bound
-#' # ------------------------------------------------- #
-#' # Original design
-#' x <- gs_design_ahr(
-#'   enroll_rate = enroll_rate, fail_rate = fail_rate,
-#'   alpha = alpha, beta = beta, ratio = ratio,
-#'   info_scale = "h0_info",
-#'   info_frac = NULL, analysis_time = c(20, 36),
-#'   upper = gs_spending_bound,
-#'   upar = list(sf = sfLDOF, total_spend = alpha, timing = c(20, 36) / 36),
-#'   test_upper = TRUE,
-#'   lower = gs_spending_bound,
-#'   lpar = list(sf = sfLDOF, total_spend = beta, timing = c(20, 36) / 36),
-#'   test_lower = c(TRUE, FALSE),
-#'   binding = FALSE) |> to_integer()
+#'   ustime = c(min(190, planned_event_ia) / planned_event_fa, 1),
+#'   lstime = c(min(190, planned_event_ia) / planned_event_fa, 1),
+#'   event_tbl = data.frame(analysis = c(1, 1, 2, 2),
+#'                          event = c(50, 140, 50, 250)))
 #'
-#' # Updated design due to potential change of multiplicity graph
+#' # Alpha is updated to 0.05
 #' gs_update_ahr(x = x, alpha = 0.05)
 gs_update_ahr <- function(
     x = NULL,
     alpha = NULL,
     ustime = NULL,
     lstime = NULL,
-    observed_data = NULL) {
+    event_tbl = NULL) {
```

Comment:
When I recommended only adding `event_tbl`, I didn't know that `observed_data` was already an option. Is this a common use case? I worry about backwards compatibility.

One option would be to start by supporting both `observed_data` and `event_tbl`, but emit a deprecation notice if a user provides `observed_data`. Then in a future release, we could remove `observed_data` completely to simplify the code.

Reply:
Something like the following:

```R
if (!is.null(observed_data)) {
  warning(
    "The argument observed_data is deprecated and will be removed in a future release.\n",
    "Please update your code to use the argument event_tbl."
  )
  event_tbl <- func_to_convert(observed_data)
}
```

Reply:
But this is ultimately your call. I don't know how common it was to use `observed_data`

Reply:
Thank you for your suggestions! In some situations, we may use the entire TTE dataset as `observed_data`. However, I've noticed that using only `event_tbl` without `observed_data` seems to be more prevalent in practice.

Considering this, I decided to focus on only taking`event_tbl`. Additionally, I will create a function to convert `observed_data` into `event_tbl`, ensuring that we maintain flexibility in data handling.

Please let me know if you have any further thoughts or suggestions on this approach!